protocols/gossipsub/src/peer_score.rs

```diff
@@ -358,8 +374,14 @@ impl PeerScore {
             let excess = peer_stats.slow_peer_penalty - self.params.slow_peer_threshold;
             score += excess * self.params.slow_peer_weight;
         }
-
-        score
+        #[cfg(feature = "metrics")]
+        return (
+            score,
+            num_message_deficit_penalties,
+            num_ip_colocation_penalties,
+        );
+        #[cfg(not(feature = "metrics"))]
+        return (score, 0, 0);
```

Comment:
See above comment. Don't think we need to feature gate here.

Reply:
Okay, these do look unnecessarily verbose.